src/internal/ppsdb/ppsdb.go

```diff
@@ -26,7 +55,23 @@ func Pipelines(db *sqlx.DB, listener col.PostgresListener) col.PostgresCollectio
 		db,
 		listener,
 		&pps.PipelineInfo{},
-		nil,
+		pipelinesIndexes,
+		col.WithKeyGen(func(key interface{}) (string, error) {
+			if commit, ok := key.(*pfs.Commit); ok {
+				if commit.Branch.Repo.Type != pfs.SpecRepoType {
```

Comment:
This should be refactored into a helper function that takes a pipeline name and spec commit, and returns the key. Anything that accesses the pipeline collection through the name & commit key should use the helper function.

Reply:
Actually, since you are just relying on the spec repo name, then I guess it just needs to take a spec commit. I guess that explains why you create the key explicitly below since you don't have the spec commit, just the id. It would probably be better to get the spec commit and pass that to a key helper function rather than having the key defined in multiple places.

Reply:
Okay, so I was not aware that you could pass in either a string or a proto to the collection package, and it would attempt to handle both. That seems like something that could end up being too clever.

Reply:
I still think we should have a key function that takes a spec commit, and that would be the only way to get the string encoding.